### ISSUE TITLE
Ready event

### ISSUE THREAD

**Issue by nbarbettini** [okta/okta-signin-widget#539]
### I'm submitting a
  - feature request

### Background info

When `renderEl()` is called, the Widget is not always immediately ready for input. If you provide a state token, for example, the Widget first calls the Authentication API to introspect the token.

To make it easy to execute code when the Widget is ready for user input, we should add a `ready` event that is emitted when initial loading is done and the Widget is waiting for user input.

**Comment by kevlened** [comment]
Hey Nate! I wonder if this can already be achieved by:

```js
signIn.on('pageRendered', function (data) {
  // Every route calls this function
});
```

**Comment by nbarbettini** [comment]
@kevlened Yep, that would work. However, that function would be called every time the Widget transitioned to a new page.

I'm thinking of a `ready` event to function effectively like:

```
signIn.once('pageRendered', function() {
  // ...
});
```

**Comment by kevlened** [comment]
In that case, does it make sense to expose a more general `once` method like a traditional event emitter?

**Comment by jmelberg-okta** [comment]
Resolved with #577, and will be released in `2.15.0` in the coming weeks.
